disnake/channel.py

```diff
@@ -477,13 +479,132 @@ async def edit(
+        topic: Optional[:class:`str`]
+            The topic of the new channel. If not provided, defaults to this channel's topic.
+        position: :class:`int`
+            The position of the new channel. If not provided, defaults to this channel's position.
+        nsfw: :class:`bool`
+            Whether the new channel should be marked as NSFW. If not provided, defaults to this channel's NSFW value.
+
+            .. note::
+                If the Guild doesn't have the ``COMMUNITY`` feature enabled this will defaults to this :attr:`TextChannel.type`.
+
+        category: Optional[:class:`abc.Snowflake`]
+            The category where the new channel should be grouped. If not provided, defaults to this channel's category.
+        slowmode_delay: :class:`int`
+            The slowmode of the new channel. If not provided, defaults to this channel's slowmode.
+        default_thread_slowmode_delay: Optional[:class:`int`]
+            Specifies the slowmode rate limit at which users can send messages
+            in newly created threads in this channel, in seconds.
+            This does not apply retroactively to existing threads.
+            A value of ``0`` or ``None`` disables slowmode. The maximum value possible is ``21600``. If not provided, defaults
+            to this channel's default thread slowmode delay.
+        default_auto_archive_duration: Union[:class:`int`, :class:`ThreadArchiveDuration`]
+            The default auto archive duration of the new channel. If not provided, defaults to this channel's default auto archive duration.
+        overwrites: :class:`Mapping`
+            A :class:`Mapping` of target (either a role or a member) to :class:`PermissionOverwrite`
+            to apply to the channel. If not provided, defaults to this channel's overwrites.
+        news: :class:`bool`
+            Whether the new channel should be a news channel. News channels are text channels that can be followed.
+            This is only available to guilds that contain ``NEWS`` in :attr:`Guild.features`. If not provided, defaults to the current type of this channel.
+        reason: Optional[:class:`str`]
+            The reason for cloning this channel. Shows up on the audit log.
+
+        Raises
+        ------
+        Forbidden
+            You do not have the proper permissions to create this channel.
+        HTTPException
+            Creating the channel failed.
+
+        Returns
+        -------
+        :class:`TextChannel`
+            The newly created text channel.
+        """
+        if news is not MISSING and "COMMUNITY" in self.guild.features:
+            # if news is True set the channel_type to News, otherwise if it's False set it to Text
+            channel_type = ChannelType.news if news else ChannelType.text
+        else:
+            # if news is not given falls back to the original TextChannel type
+            channel_type = self.type
```

Comment:
what error does the API provide if news is provided and the guild doesn't have the feature? This currently is not intuitive.

Reply:
50035, "Invalid Form Body"

Reply:
IMO it is fine to not gate this. We don't gate this on other methods right now, like `Guild.create_text_channel(news=True, ...)`

Reply:
The main problem here was that a normal user wouldn't understand the error since the Api returns a list of allowed channel types as numbers, hence I thought we could do something since it's not that complex (maybe unexpected right now)

Reply:
Its inconsistent with the rest of the library.

Reply:
ok I'll delete it 😔